packages/core/status/core-status-server-internal/src/status_service.ts

```diff
@@ -149,19 +162,6 @@ export class StatusService implements CoreService<InternalStatusServiceSetup> {
       ...commonRouteDeps,
     });
 
-    if (commonRouteDeps.config.allowAnonymous) {
-      http.registerPrebootRoutes('', (prebootRouter) => {
-        registerStatusRoute({
-          router: prebootRouter,
-          ...commonRouteDeps,
-          config: {
-            ...commonRouteDeps.config,
-            allowAnonymous: true,
-          },
-        });
-      });
-    }
```

Comment:
TIL we were actually registering the default status handler on the preboot server when `status.allowAnonymous` was `true`. I removed this logic to plug the static handler we discussed about. 

I could eventually register the dynamic handler instead, but I feel like it's not really necessary (the static `unavailable` response seems sufficient during preboot given that's the only status the server can have anyway)